apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
+      assert.ok(tm.hide.calledOnce);
     });
 
-    test('all apps should leaveTaskManager', function() {
-      MockStackManager.mStack.forEach(function(app) {
-        this.sinon.spy(app, 'leaveTaskManager');
-      }, this);
-
-      taskManager.hide();
-
-      MockStackManager.mStack.forEach(function(app) {
-        sinon.assert.calledOnce(app.leaveTaskManager);
-      }, this);
+    test('hit "home" while already closing, ignore event (bug 1203772)', () => {
+      sinon.spy(tm, 'hide');
+      tm.hide();
+      // If we receive a "home" event while already going home, we should
+      // ignore it. (i.e. in this implementation, _isTransitioning is true)
+      var result = tm.respondToHierarchyEvent(new CustomEvent('home'));
+      assert.equal(result, true);
+      assert.ok(tm.hide.calledOnce);
     });
 
-    test('hide: raises cardviewclosed event', function(done) {
-      taskManager.newStackPosition = 1;
-      waitForEvent(window, 'cardviewclosed').then(function(event) {
-        assert.equal(typeof event.detail, 'object',
-                    'gets event with detail object');
-        assert.equal(event.detail.newStackPosition, 1,
-                    'event detail reflects taskManager.newStackPosition');
-        delete taskManager.newStackPosition;
-      }, failOnReject).then(done, done);
-      taskManager.hide();
-      this.sinon.clock.tick(101);
+    test('click when empty takes you home', () => {
+      sinon.spy(tm, 'hide');
+      tm.handleEvent(new CustomEvent('click'));
+      clock.tick(1000);
+      assert.ok(tm.hide.calledOnce);
     });
 
-    test('hide: removes cards', function(done) {
-      waitForEvent(window, 'cardviewclosed').then(function(event) {
-        assert.equal(cardsList.childNodes.length, 0,
-                    'all card elements are gone');
-        assert.equal(Object.keys(taskManager.cardsByAppID).length, 0,
-                    'cards lookup is empty');
-        done();
-      }, failOnReject);
-      taskManager.hide();
-      this.sinon.clock.tick(101);
-    });
-
-    test('hide: calls card.destroy', function(done) {
-      var firstCard = taskManager.getCardAtIndex(0);
-      var secondCard = taskManager.getCardAtIndex(1);
-      var destroyStub1 = sinon.stub(firstCard, 'destroy');
-      var destroyStub2 = sinon.stub(secondCard, 'destroy');
-
-      waitForEvent(window, 'cardviewclosed').then(function(event) {
-        assert.isTrue(destroyStub1.calledOnce,
-                      '1st card.destroy was called once');
-        assert.isTrue(destroyStub2.calledOnce,
-                      '2nd card.destroy was called once');
-        destroyStub1.restore();
-        destroyStub2.restore();
-        done();
-      }, failOnReject);
-      taskManager.hide();
-      this.sinon.clock.tick(101);
+    test('takes you home after closing all cards', (done) => {
+      sinon.spy(tm, 'hide');
+      tm.handleEvent(new CustomEvent('appterminated'));
```

Comment:
same, dispatched from the app's frame, you might need to move this test to the "With Populated Task Manager" suite (which is more realistic too)